### PR TITLE
adv_service_data

### PR DESCRIPTION
**Pull request by jsotogaviard**
Pull Request Guidelines for Bleak
---------------------------------

Before you submit a pull request, check that it meets these guidelines:

1. If the pull request adds functionality, the docs should be updated.
2. Modify the `CHANGELOG.rst`, describing your changes as is specified by the
   guidelines in that document.
3. The pull request should work for Python 3.6+ on the following platforms:
    - Windows 10, version 16299 (Fall Creators Update) and greater
    - Linux distributions with BlueZ >= 5.43
    - OS X / macOS >= 10.11
4. Squash all your commits on your PR branch, if the commits are not solving
   different problems and you are committing them in the same PR. In that case,
   consider making several PRs instead.
5. Feel free to add your name as a contributor to the `AUTHORS.rst` file!

### CODE REVIEW

File: bleak/backends/corebluetooth/scanner.py [hunk 1]
```diff
@@ -132,6 +132,11 @@ async def get_discovered_devices(self) -> List[BLEDevice]:
                 for u in advertisementData.get("kCBAdvDataServiceUUIDs", [])
             ]
 
+            service_data = {}
+            adv_service_data = advertisementData.get("kCBAdvDataServiceData", [])
+            for u in adv_service_data:
+                service_data[cb_uuid_to_str(u)]= bytes(adv_service_data[u])
```
Comment (dlech) [inline bleak/backends/corebluetooth/scanner.py:138]: please add a space before the equals to fix the format test failure

**Review by dlech — CHANGES_REQUESTED** [review]
Thanks! I'm not sure how we missed this before!

File: CHANGELOG.rst [hunk 2]
```diff
@@ -10,6 +10,11 @@ and this project adheres to `Semantic Versioning <https://semver.org/spec/v2.0.0
 `Unreleased`_
 -------------
 
+Added
+~~~~~
+
+* Added ``AdvertisementServiceData`` in BLEDevice.
```
Comment (dlech) [inline CHANGELOG.rst:16]: Please add something to say that this is just for macOS here.

File: bleak/backends/corebluetooth/scanner.py (hunk 1 above)
Comment (jsotogaviard) [reply bleak/backends/corebluetooth/scanner.py]: done

File: CHANGELOG.rst (hunk 2 above)
Comment (jsotogaviard) [reply CHANGELOG.rst]: done

**Review by dlech — APPROVED** [review]

**Comment by dlech** [comment]
Thanks for updating. Is it safe to assume you have tested this with a device that provides service data in the advertising data?

**Comment by jsotogaviard** [comment]
I have tested with xiaomi mi thermometers and atc fw : https://github.com/atc1441/ATC_MiThermometer

**Comment by hbldh** [comment]
@jsotogaviard If you could use your device as an example to try to implement this for the backend missing this data (the BlueZ/Linux backend I think), that would be great!

Regardless, thank you for the contribution!

**Comment by jsotogaviard** [comment]
I only Have Mac/windows computers ...

**Comment by dlech** [comment]
>  If you could use your device as an example to try to implement this for the backend missing this data (the BlueZ/Linux backend I think), that would be great!

This was actually implemented in #334 and I tested all backends then. But somehow we managed to delete this from macOS in the meantime.

**Pull request merged**